### DISCUSSION TITLE
How to structure pages in a headless cms

### DISCUSSION THREAD

**Discussion by pm-designs** [directus/directus#3813] [Feature Requests]
I like the new frameworks and Headless CMS, but i wonder how the content-creator can structure their pages. A normal Websites have different pages and sections with a different structure of content (Header, Text, Slider, Teaser-row, Text, Gallery, Footer). Is this not an essential need for the most webpages and CMS? In my point of view with a headless CMS nowdays the developer gives a very static template to the content-creator with no structure options.

Am i right or is there any solution for this problem?

**Comment by erkand-imeri** [comment]
The content-creator generates an API (that's the end-point where it cares) and it gives freedom to the front-end to structure their pages however they want.

This is a very powerful concept in fact. Especially for highly customizable software needs.

**Reply by pm-designs** [reply]
Thank you for your answer.
With content-creator i mean normal people who use directus for example as cms. (None Developer)
I know what you mean with api and structured JSON. But how can i save the dynamically ordered structure of different pages in directus?

I know there are decoupled CMS. Is this the only way i can do that or is it possible wit directus too?

For example my page is develop with vue, Nuxtjs and directus. The structure of the page is hard defined in the code. So with directus i can not change the order of the sections.

http://madeby.pm

**Comment by benhaynes** [comment]
What @erkand-imeri said is correct, but what you're looking for is the Many-to-Any (M2A) relationship... or "page builder" interface.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
